Platform/obc/hal/comms_hal.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file comms_hal.c
+ * @author Arash Yazdani
+ * @date 2020-10-01
+ */
+
+/*Once the hardware is connected, toggle this Macro*/
+#define Stubbed = 1;
+
+#include "obc_hal.h"
+#include "comms_hal.h"
+#include <stddef.h>
+#include <FreeRTOS.h>
+#include <csp/csp.h>
+#include <csp/csp_endian.h>
+#include <TempSensor/TempSensor.h>
+#include "hal.h"
+#include "services.h"
+#include "queue.h"
+
+
+/* This function is kept for test purposes. Will be deleted */
+void HAL_comm_getTemp(uint32_t *sensor_temperature) {
+  #ifdef Stubbed
+    *sensor_temperature = 18;
+  #else
+    HAL_get_temperature(sensor_temperature);
+  #endif
+}
+
+/*Comments for the PR reviewer:
+ * 1- STX functions return value. Should we use them for the
+ * SAT_returnState in communication_service_app?
+ * 2- Like RTC we can check if the value is stored after the
+ * set functions (change the get functions).
+ * But there is no point when eventually Stubbed will be 0.
+ * 3- Do we need comments for each function?
+ */
+
+void HAL_S_getFreq (uint32_t *S_freq) { //should be replaced with float
+  #ifdef Stubbed
+    *S_freq = 2210.5;
+  #else
+    STX_getFrequency(*S_freq);
+  #endif
+};
+
+void HAL_S_getpaPower (uint32_t *S_paPower) {
+  #ifdef Stubbed
+    *S_paPower = 2;
+  #else
+    STX_getPaPower(*S_paPower);
+  #endif
+};
+
+void HAL_S_getControl (uint32_t *S_paStatus, uint32_t *S_paMode) { //must be uint8-t
+  #ifdef Stubbed
+    *S_paStatus = 1;
+    *S_paMode = 3;
+  #else
+    STX_getControl(*S_paStatus, *S_paMode);
+  #endif
+};
+
+void HAL_S_getEncoder (uint8_t *S_enc_scrambler, uint8_t * S_enc_filter, uint8_t * S_enc_mod, uint8_t * S_enc_rate){
+  #ifdef Stubbed
+    *S_enc_scrambler = 0;
+    *S_enc_filter = 0;
+    *S_enc_mod = 0;
+    *S_enc_rate = 0;
+  #else
+    STX_getEncoder(uint8_t * S_enc_scrambler, uint8_t * S_enc_filter, uint8_t * S_enc_mod, uint8_t * S_enc_rate);
+  #endif
+}
+
+void HAL_S_getFirmwareV (float *S_Firmware_Version){
+  #ifdef Stubbed
+    *S_Firmware_Version = 7.14;
+  #else
+    STX_getFirmwareV (*S_Firmware_Version);
+  #endif
+}
+
+void HAL_S_getStatus (uint8_t * S_PWRGD, uint8_t *S_TXL){
```

Comment:
Functions prototypes should take the structures as arguments I think

Reply:
I had some problems with this. Now that I was about to explain it, I think I figured out what the problem was haha. I'll try again.

Reply:
Ok. It's working now (using arrow operation instead of dot). However, the EH functions are not compatible with the struct arguments. So, we have two options: Keep them as is, Or if it's not stubbed, we send the member of the structures to the EH functions (not sure if it'd work...might get some errors since we can't test them now)

`STX_return HAL_S_getControl (Sband_PowerAmplifier *S_PA) {
  #ifdef Stubbed
    S_PA->status = 1;
    S_PA->mode = 3;
    return IS_STUBBED;
  #else
    return STX_getControl(*S_PA.status, *S_PA.mode);
  #endif
};`

Summary: Stubbed is working, not sure if this format of assigning arguments to STX would work. (Sorry that it's for a different function. I started from the top and I'm setting a standard to follow for the rest)

Reply:
I think that's ok, but `return STX_getControl(*S_PA.status, *S_PA.mode);` should probably be `return STX_getControl(&S_PA->status, &S_PA->mode);`